### PR TITLE
Enhance SQLNodeConverterEngine to support 'select_not_in_with_single_table'

### PR DESCRIPTION
**Pull request by cardigan1008**
Ref #26871 .

Changes proposed in this pull request:
  - Finish task 4: select_not_in_with_single_table
  - dbtypes: MySQL, openGauss, PostgreSQL

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: test/it/optimizer/src/test/resources/converter/select-table.xml [hunk 1]
```diff
@@ -18,4 +18,8 @@
 
 <sql-node-converter-test-cases>
     <test-cases sql-case-id="select_from_with_table" expected-sql="SELECT * FROM (TABLE `t0`) AS `dt`" db-types="MySQL" />
+    <test-cases sql-case-id="select_not_in_with_single_table" expected-sql="SELECT * FROM `t_order_item` WHERE `item_id` IS NOT NULL AND `item_id` NOT IN (?, ?) ORDER BY `item_id`" db-types="MySQL" sql-case-types="PLACEHOLDER" />
```
Comment (strongduanmu) [inline test/it/optimizer/src/test/resources/converter/select-table.xml:21]: Why only test LITERAL?

**Review by strongduanmu — CHANGES_REQUESTED** [review]

**Comment by strongduanmu** [comment]
> Fixes #26871 .

@cardigan1008 Please use ref #26871

File: test/it/optimizer/src/test/resources/converter/select-table.xml (hunk 1 above)
Comment (cardigan1008) [reply test/it/optimizer/src/test/resources/converter/select-table.xml:21]: Thank you for your feedback. I will work on adding tests for PLACEHOLDER as soon as possible and update the PR accordingly.

File: test/it/optimizer/src/test/resources/converter/select-table.xml [hunk 2]
```diff
@@ -17,5 +17,9 @@
   -->
 
 <sql-node-converter-test-cases>
-    <test-cases sql-case-id="select_from_with_table" expected-sql="SELECT * FROM (SELECT * FROM `t0`) AS `dt`" db-types="MySQL" />
+    <test-cases sql-case-id="select_from_with_table" expected-sql="SELECT * FROM (TABLE `t0`) AS `dt`" db-types="MySQL" />
```
Comment (strongduanmu) [inline test/it/optimizer/src/test/resources/converter/select-table.xml:20]: @cardigan1008 Plaese merge master branch, I modify this expected-sql.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

**Comment by strongduanmu** [comment]
Hi @cardigan1008, can you merge master branch to keep to keep the code up to date.

**Comment by cardigan1008** [comment]
> Hi @cardigan1008, can you merge master branch to keep to keep the code up to date.

Done.

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**